source/docs/software/frc-glossary.rst

```diff
@@ -11,21 +11,51 @@ FRC Glossary
    auto
       The first phase of each match is called Autonomous (auto) and consists of the robot's running pre-programmed instructions.
 
+   call stack
+      A specially-organized region of memory which helps the program keep track of what function it is in, and store the values of some variables. See `call stack <https://en.wikipedia.org/wiki/Call_stack>`__ on wikipedia for more info.
+
+   Classical Mechanics
+      The branch of physics which studies and describes the motion of relatively large, relatively slow objects. See `Classical Mechanics <https://en.wikipedia.org/wiki/Classical_mechanics>`__ on wikipedia for more info.
+
    COTS
       Commercial off the shelf, a standard (i.e. not custom order) part commonly available from a vendor to all teams for purchase.
 
+   compositions
+      A formal software term a set of multiple objects (such as commands) grouped into a bigger object. See `object composition <https://en.wikipedia.org/wiki/Object_composition>`__ on wikipedia for more info.
+
+   CRTP
+      Continuously Recurring Template Pattern - A software idiom in which a class `X`` derives from a class template instantiation using `X`` itself as a template argument. See `CRTP <https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern>`__ on wikipedia for more info.
+
    C++
       One of the three officially supported programming languages.
 
+   declarative programming
+      A style of software which focuses on describing *what* a program should do, rather than *how* it gets done. See `declarative programming <https://en.wikipedia.org/wiki/Declarative_programming>`__ on wikipedia for more info.
+
+   dependency injection
+      A software design pattern where each class receives all objects it depends upon as arguments to its constructor. See `dependency injection <https://en.wikipedia.org/wiki/Dependency_injection>`__ on wikipedia for more info.
```

Comment:
minus "through its constructor."  Constructor injection is one type of DI, but there are others.